workflow.go

```diff
@@ -625,3 +626,43 @@ const DefaultVersion Version = -1
 func GetVersion(ctx Context, changeID string, minSupported, maxSupported Version) Version {
 	return getWorkflowEnvironment(ctx).GetVersion(changeID, minSupported, maxSupported)
 }
+
+// SetQueryHandler sets the query handler to handle workflow query. The queryType specify which query type this handler
```

Comment:
I believe that we discussed that query function is executed outside of dispatcher context. We have somehow document that it cannot use any workflow blocking functions like channel.get, etc.

Reply:
Will update comments to point that out. However, if user do try to use Channel.Get, the code will block indefinitely. We probably need to error out in that case, meaning that we need to detect if those function is called outside of a workflow context. I will address that in a separate CR.